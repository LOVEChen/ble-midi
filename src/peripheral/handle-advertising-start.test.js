/* eslint-env jest */
jest.mock('../util/logger');

const handleAdvertisingStart = require('./handle-advertising-start').default;
let mockLog;

beforeEach(() => {
  jest.resetAllMocks();
  mockLog = require('../util/logger').default;
});

test('should handleAdvertisingStart success', () => {
  const bleno = jest.fn();
  bleno.setServices = jest.fn();
  handleAdvertisingStart(bleno, null);
  expect(mockLog.mock).toMatchSnapshot();
  expect(bleno.setServices.mock).toMatchSnapshot();
});

test('should handleAdvertisingStart error', () => {
  const bleno = jest.fn();
  handleAdvertisingStart(bleno, new Error('mock-error'));
  expect(mockLog.mock).toMatchSnapshot();
});
