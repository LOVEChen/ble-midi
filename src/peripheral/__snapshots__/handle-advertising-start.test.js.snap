// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handleAdvertisingStart error 1`] = `
Object {
  "calls": Array [
    Array [
      "advertisingStart: Error: mock-error",
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should handleAdvertisingStart success 1`] = `
Object {
  "calls": Array [
    Array [
      "advertisingStart: success",
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should handleAdvertisingStart success 2`] = `
Object {
  "calls": Array [
    Array [
      Array [
        PrimaryService {
          "characteristics": Array [
            Characteristic {
              "_events": Object {
                "indicate": [Function],
                "notify": [Function],
                "readRequest": [Function],
                "subscribe": [Function],
                "unsubscribe": [Function],
                "writeRequest": [Function],
              },
              "_eventsCount": 6,
              "descriptors": Array [],
              "onIndicate": [Function],
              "onNotify": [Function],
              "onReadRequest": [Function],
              "onSubscribe": [Function],
              "onUnsubscribe": [Function],
              "onWriteRequest": [Function],
              "properties": Array [
                "read",
                "write",
                "writeWithoutResponse",
                "notify",
              ],
              "secure": Array [
                "read",
                "write",
                "writeWithoutResponse",
                "notify",
              ],
              "uuid": "7772E5DB38684112A1A9F2669D106BF3",
              "value": null,
            },
          ],
          "uuid": "03B80E5AEDE84B33A7516CE34EC4C700",
        },
      ],
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;
